src/ir/template.rs

```diff
@@ -34,6 +34,22 @@ use super::derive::CanDeriveCopy;
 use super::item::Item;
 use super::traversal::{EdgeKind, Trace, Tracer};
 
+/// A trait for things which may or may not be a named template type parameter.
+pub trait AsNamed {
```

Comment:
I kind of want to do a renaming pass "named" -> "type parameter" everywhere as a follow up. What do you think of that instead? (So this would become `AsTypeParameter` or `AsTypeParam`).